File: src/main/java/com/floorestimatepro/export/EstimateExporter.java
package com.floorestimatepro.export;

import java.nio.file.Files;
import java.nio.file.Path;

import com.floorestimatepro.model.EstimateResult;

public class EstimateExporter{
    public static String toCsv(EstimateResult result){
        return "netPixelArea,netRealArea,materialSqFt,estimatedCost\n"
            + result.netPixelArea() + ","
            + result.netRealArea() + ","
            + result.materialSqFt() + ","
            + result.estimatedCost() + "\n";
    }

    public static void exportCsv(EstimateResult result, String filePath) throws Exception{
        String csv = toCsv(result);
        Files.writeString(Path.of(filePath), csv);
    }
}

File: python/src/floor_estimate_pro/export/estimate_exporter.py
from __future__ import annotations

from pathlib import Path

from floor_estimate_pro.model.estimate_result import EstimateResult


class EstimateExporter:
    @staticmethod
    def to_csv(result: EstimateResult) -> str:
        return (
            "netPixelArea,netRealArea,materialSqFt,estimatedCost\n"
            + str(result.net_pixel_area) + ","
            + str(result.net_real_area) + ","
            + str(result.material_sq_ft) + ","
            + str(result.estimated_cost) + "\n"
        )

    @staticmethod
    def export_csv(result: EstimateResult, file_path: str) -> None:
        csv = EstimateExporter.to_csv(result)
        Path(file_path).write_text(csv, encoding="utf-8")
